taskrunner/host_gateway.go

```diff
@@ -168,6 +163,12 @@ func (agbh *AgentHostGateway) prepRemoteHost(hostObj host.Host, sshOptions []str
 	// run the setup script with the agent
 	if logs, err := hostutil.RunSSHCommand("setup", hostutil.SetupCommand(&hostObj), sshOptions, hostObj); err != nil {
 		event.LogProvisionFailed(hostObj.Id, logs)
+		grip.Errorf("error running setup script on %s", hostObj.Id)
+		// there is no guarantee setup scripts are idempotent, so we terminate the host if the setup script fails
+		if err := hostObj.DisablePoisonedHost(); err != nil {
+			return "", errors.Wrapf(err, "error terminating host %s", hostObj.Id)
```

Comment:
If we can't terminate a host that failed setup, is that something that should we should be grip logging at a fairly high level? It seems like something we'd want a notification for

Reply:
All the method does is change some data in the DB and the monitor will terminate it later... 

It's a fair point that if we get to line 166, it'd be good to log that in some way, but the truth is that its basically never actionable.